activestorage/lib/active_storage/service/s3_service.rb

```diff
@@ -156,10 +159,12 @@ def stream(key)
         chunk_size = 5.megabytes
         offset = 0
 
-        raise ActiveStorage::FileNotFoundError unless object.exists?
+        raise ActiveStorage::FileNotFoundError unless object.exists?(download_options)
 
-        while offset < object.content_length
```

Comment:
`object.content_length` fails in the case of SSE-C, so I had to refactor this to use `object.head.content_length`.